service/history/engine/engineimpl/query_workflow.go

```diff
@@ -87,7 +87,7 @@ func (e *historyEngineImpl) QueryWorkflow(
 	}
 	deadline := time.Now().Add(queryFirstDecisionTaskWaitTime)
 	for mutableStateResp.GetPreviousStartedEventID() <= 0 && time.Now().Before(deadline) {
-		<-time.After(queryFirstDecisionTaskCheckInterval)
+		time.Sleep(queryFirstDecisionTaskCheckInterval)
```

Comment:
hmmm.  can't cancel this when shutting down 🤔 

any idea what the time-scale of this sleep is?  would it be worth trying to make it cancelable immediately, or is it like <100ms and doesn't matter much?

Reply:
it's 200ms

Reply:
well.  might be too risky to try to fix right now / as a "free" addition to what is otherwise a refactor.

and `time.Sleep` is a clear sign of "this should probably use context" for later fixing, so it won't be any harder to find.

sgtm, `time.Sleep(..)` works 👍